quota/etcd/storagepb/storagepb.proto

```diff
@@ -0,0 +1,86 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+// Package storagepb contains definitions for quota storage protos, which are recorded in etcd.
+package storagepb;
+
+// Data contained in a quota bucket.
+// Stored at each each quota's zero bucket. For example,
+// quotas/global/read/0 or quotas/trees/$id/read/0.
+message Bucket {
+
+  // Number of tokens left in the bucket.
+  int64 tokens = 1;
+
+  // Timestamp of the last time the bucket got replenished.
+  int64 last_replenish_millis_since_epoch = 2;
+}
+
+// Configuration for all quotas.
+// Stored at quotas/configs.
+message Configs {
+
+  // Known quota configurations.
+  repeated Config configs = 1;
```

Comment:
I thought about this, but there's room for mistakes on both representations. The list version may contain duplicates, the map version may have mismatches between the map key and Config.Name. Ultimately, the concept I have in mind is better served by the list version.

I've added tests / validation for duplicate names.